sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/_tests_/ece-response-integration.scheduler.e2e-spec.ts

```diff
@@ -558,10 +591,292 @@ describe(
+        "Disbursement 1119353191, record failed to process due to reason: Invalid enrolment confirmation flag.",
+      ];
+      expect(processResult).toStrictEqual([expectedResult]);
+      // Expect the delete method to be called.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+    });
+
+    it("Should skip the processing and log error when detail record with invalid enrolment confirmation date and pay to school amount is present and process other disbursements.", async () => {
+      // Arrange
+      // Including a valid disbursement in this test case to ensure that
+      // when there is a enrolment data validation error, only that particular disbursement is aborted
+      // and other disbursements are processed.
+      // Enable integration for institution location
+      // used for test.
+      await enableIntegration(locationCONF, db);
+      const confirmEnrolmentResponseFile = path.join(
+        process.env.INSTITUTION_RESPONSE_FOLDER,
+        locationCONF.institutionCode,
+        ECE_RESPONSE_FILE_NAME,
+      );
+
+      // Create disbursement to confirm enrolment.
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        undefined,
+        {
+          applicationStatus: ApplicationStatus.Enrolment,
+        },
+      );
+
+      const [disbursement] =
+        application.currentAssessment.disbursementSchedules;
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Modify the data in mock file to have the correct values for
+      // disbursement and application number.
+      mockDownloadFiles(
+        sftpClientMock,
+        [ECE_RESPONSE_FILE_NAME],
+        (fileContent: string) => {
+          return fileContent
+            .replace("DISBNUMBER", disbursement.id.toString().padStart(10, "0"))
+            .replace("APPLNUMBER", application.applicationNumber)
+            .replace("ENRLDATE", formatDate(new Date(), "YYYYMMDD"))
+            .replace("Y20230418N000000", "YNOTADATENNANNUM");
+        },
+      );
+
+      // Act
+      const processResult = await processor.processECEResponse(job);
+
+      // Assert
+      const expectedResult: ProcessSummaryResult = new ProcessSummaryResult();
+      expectedResult.summary = [
+        `Starting download of file ${confirmEnrolmentResponseFile}.`,
+        `Disbursement ${disbursement.id}, enrolment confirmed.`,
+        `The file ${confirmEnrolmentResponseFile} has been deleted after processing.`,
+        "Total file parsing errors: 0",
+        "Total disbursements found: 2",
+        "Disbursements successfully updated: 1",
+        "Disbursements skipped to be processed: 0",
+        "Disbursements considered duplicate and skipped: 0",
+        "Disbursements failed to process: 1",
+      ];
+      expectedResult.errors = [
+        "Disbursement 1119353191, record failed to process due to reason: Invalid enrolment confirmation date, Invalid pay to school amount.",
```

Comment:
since the test is checking for `Invalid pay to school amount.`. Can you add it in `mockDownloadFiles`, it will be easier for a developer to identify, it instead going to test and counting the index

Reply:
> Good work @dheepak-aot added some comments.
> 
> I missed to ask this in your previous PR, are we not considering/testing the below scenarios, ![image](https://user-images.githubusercontent.com/77353155/244523431-39e4e6ea-ee6f-4dd6-8e9c-bc8c5d66ea2d.png)

Are we not considering below scenarios - No. We are considering them.

Are we not including them in test/cases - So far we didn't but we can include in upcoming PR. When we did #1501  Please remember that the expectation of that ticket is to cover some basic scenarios e2e tests to to cover every possible scenario. But because we are expanding the E2E tests, this ticket can do that.